tests/check_codecov

```diff
@@ -0,0 +1,172 @@
+        trap 'cleanup ; wait ; exit 1 ;' SIGTERM || EXIT_CODE=4
+        trap 'cleanup ; wait ; exit 1 ;' SIGQUIT || EXIT_CODE=5
+        trap 'cleanup ; wait ; exit 1 ;' SIGSTOP || EXIT_CODE=7
+        trap 'cleanup ; wait ; exit 1 ;' SIGINT || EXIT_CODE=8
+        trap 'cleanup ; wait ; exit 1 ;' SIGABRT || EXIT_CODE=9
+        trap 'cleanup ; wait ; exit ${EXIT_CODE} ;' EXIT || EXIT_CODE=1
+else
+        echo CodeCov already in progress by `head ${LOCK_FILE}` ;
+        false ;
+        exit 126 ;
+fi
+
+# this is how test files are found:
+
+# THIS IS THE ACTUAL TEST DIR USED (update _TEST_ROOT_DIR as needed)
+_TEST_ROOT_DIR="./" ;
+if [[ -d ./multicast ]] ; then
+	_TEST_ROOT_DIR="./multicast" ;
+	_TEST_ROOT_DIR="./" ;
+elif [[ -d ./tests ]] ; then
+	_TEST_ROOT_DIR="./" ;
+else
+	echo "FAIL: missing valid folder or file"
+	EXIT_CODE=1
+fi
+
+# This File  MUST BE GIT-IGNORED to be SAFELY USED to store Tokens and env vars (update logic as needed)
+if [[ ( -r ./codecov_env ) ]] ; then
+	source ./codecov_env 2>/dev/null || true ;
+fi
+
+if [[ ( -r ./codecov.yml ) ]] ; then
+	cat codecov.yml | curl --data-binary @- https://codecov.io/validate || EXIT_CODE=6
+fi
+if [[ ( -r ./.codecov.yml ) ]] ; then
+	cat ./.codecov.yml | curl --data-binary @- https://codecov.io/validate || EXIT_CODE=6
+fi
+
+# sorry no windows support here
+if [[ $( \uname -s ) == "Darwin" ]] ; then
+	CI_OS="macos"
+else
+	CI_OS="linux"
+fi
+
+#########################
+# actual Work starts here
+#########################
+curl -fLso codecov https://uploader.codecov.io/latest/${CI_OS:-linux}/codecov ;
+for i in 1 256 512 ; do
+	curl -fLso codecov.SHA${i}SUM "https://uploader.codecov.io/latest/${CI_OS:-linux}/codecov.SHA${i}SUM" ; wait ;
+	curl -fLso codecov.SHA${i}SUM.sig "https://uploader.codecov.io/latest/${CI_OS:-linux}/codecov.SHA${i}SUM.sig" ; wait ;
+	# test sha1/sha512 signatures if found and sha256 even if not found
+	if [[ ( -r codecov.SHA${i}SUM ) ]] || [[ ( ${i} -eq 256 ) ]] ; then
+		if [[ ( -r codecov.SHA${i}SUM.sig ) ]] ; then
+			# configure your CI evironment to trust the key at https://keybase.io/codecovsecurity/pgp_keys.asc
+            # FP: 2703 4E7F DB85 0E0B BC2C 62FF 806B B28A ED77 9869
+			# OR...
+			# Set CI=true to continue on missing keys
+			gpgv codecov.SHA${i}SUM.sig codecov.SHA${i}SUM || ${CI} || EXIT_CODE=126
+			rm -vf codecov.SHA${i}SUM.sig 2>/dev/null ;
+		fi
+		shasum -a $i -c --ignore-missing codecov.SHA${i}SUM || EXIT_CODE=126
+		rm -vf codecov.SHA${i}SUM 2>/dev/null ;
+	fi
+done
+
+if [[ ( ${EXIT_CODE} -eq 0 ) ]] ; then
+	chmod -v 751 ./codecov || EXIT_CODE=126
+fi
+
+if [[ ( -x $(command -v coverage3) ) ]] ; then
+	coverage3 combine 2>/dev/null || true
+	coverage3 xml 2>/dev/null || true
+elif [[ ( -x $(command -v coverage) ) ]] ; then
+	coverage combine 2>/dev/null || true
+	coverage xml 2>/dev/null || true
+fi
+
+if [[ ( ${EXIT_CODE} -eq 0 ) ]] ; then
+	./codecov -n "Custom Test Run" -X gcov -F ${CI_OS:-linux} || EXIT_CODE=5 ;
```

Comment:
🤔 are these codes random or do they have meaning ❓
- is that meaning documented? where?